framework/tst/dslabs/framework/testing/newviz/SingleNodePanel.java

```diff
@@ -188,14 +219,15 @@ void updateState(EventTreeState s, boolean viewDeliveredMessages) {
         timerBox.removeAll();
 
         for (TimerEnvelope timer : s.timers(address)) {
-            addTimer(timer, s.canStepTimer(timer));
+            addTimer(timer, s.canStepTimer(timer),
+                     !searchDone && (settings == null || settings.deliverTimers(address)));
         }
 
         timerBox.revalidate();
         timerBox.repaint();
     }
 
-    private void addMessage(final MessageEnvelope message) {
+    private void addMessage(final MessageEnvelope message, boolean allowDelivery) {
```

Comment:
One thing to think about (possibly not for this initial PR): adding a string here that explains why delivery isn't allowed if `allowDelivery` is `false`. Then set that as the tool tip text on the button.

Reply:
Just read through your PR message properly. So, right now a message/timer isn't deliverable if we're in a pruned state or if the settings disallow delivering it. Just say one of those.


"This [message/timer] cannot be delivered because [the current state is pruned by the search]/[delivery is prohibited by the search]"

Something like that.

Reply:
Fixed in https://github.com/emichael/dslabs/pull/29/commits/c0f1eba7cc6966e25eeefa893dd43ac666d0c7a8.